orbstation/quirks/plurality.dm

```diff
@@ -0,0 +1,316 @@
+		if(!system_member_names.len)
+			to_chat(user, span_warning("ERROR: There are no members registered with [system_name]."))
+			return
+		var/clear_confirmation = tgui_alert(user, "Do you really want to clear all members from the system?", "Clear Members", list("Yes", "No"))
+		if(clear_confirmation == "Yes")
+			system_member_names.Cut()
+			to_chat(user, span_notice("Successfully cleared all members from [system_name]."))
+		return
+	if(popup_input == "Edit System Name")
+		if(system_name_locked)
+			to_chat(user, span_warning("ERROR: This chip cannot have its system name changed without being swiped first by an ID with Head of Personnel access."))
+			return
+		var/new_system_name = tgui_input_text(user, "Input the new system name:", "Edit System Name", max_length = 20)
+		if(!new_system_name)
+			return
+		to_chat(user, span_notice("Successfully changed the chip's associated system name from [system_name] to [new_system_name]."))
+		update_system_name(new_system_name)
+		return
+
+/// Changes the system_name variable and automatically updates the name of the chip to include it.
+/obj/item/plural_system_chip/proc/update_system_name(new_system_name)
+	system_name = new_system_name
+	name = "[initial(name)] {[system_name]}"
+
+/obj/item/plural_system_chip/attackby(obj/item/W, mob/user, params)
+	if(W.GetID())
+		if(allowed(user))
+			locked = !locked
+			system_name_locked = FALSE
+			balloon_alert(user, locked ? "locked" : "unlocked")
+			update_appearance()
+			return
+		else
+			balloon_alert(user, "access denied!")
+			return
+
+// Code that allows ID cards to have plural system chips attached to them and interface with them to change the label of the ID.
+
+/obj/item/card/id
+	/// Attached plural system chip, if any.
+	var/obj/item/plural_system_chip/plural_system
+	/// Whether a plural system chip can be added to the ID.
+	var/plural_system_compatible = TRUE
+
+/obj/item/card/id/attack_self(mob/user)
+	if(user.incapacitated())
+		return
+	if(!user.is_holding(src))
+		return
+	// if there is a plural system chip attached and the system has members, shows a popup through which the user can select a member to display as the current fronter
+	if(plural_system?.system_member_names.len && plural_system_compatible)
+		var/list/menu_options = plural_system.system_member_names.Copy()
+		menu_options |= "Show ID"
+		var/popup_input = tgui_input_list(user, "Select the current fronter:", "Plural System Interface", menu_options)
+		if(!popup_input)
+			return
+		if(popup_input != "Show ID")
+			plural_system.fronter_name = popup_input
+			balloon_alert(user, "[popup_input]")
+			to_chat(user, span_notice("You set [plural_system]'s current fronter to [plural_system.fronter_name]."))
+			update_label()
+			return
+	..() // if there is no plural system chip or the "Show ID" option was chosen, just show the ID like normal
+
+/obj/item/card/id/attackby(obj/item/item, mob/user, params)
+	if(istype(item, /obj/item/plural_system_chip))
```

Comment:
If this is the only item we care about you could also early return here, but that's another style preference.